### ISSUE TITLE
json2md compatible output from tokens

### ISSUE THREAD

**Issue by mmadson** [connected-web/md2json#6]
Hi there, thanks for the lib; seems to be pretty close to what I'm looking for. 

The output from tokens looks VERY very close to the input required for the json2md library. If there is an option to generate the the json in this format that would be ideal. 

Use case:

- read a markdown file into json tokens using md2json
- make some modifications, perhaps add a few new elements
- write the file back to markdown using json2md

Thanks!

**Comment by johnbeech** [comment]
Hey @mmadson - sounds fairly feasible - can I confirm that this is the json2md project you're referring to?

- https://github.com/IonicaBizau/json2md

I should be able to write some tests using json2md as a dependency, and test for json-markdown-json and markdown-json-markdown conversions.

**Comment by mmadson** [comment]
@johnbeech yup, exactly right, the sample in their readme shows the expected input format I believe this performs the necessary transformation:

```
const json2mdInputFormat = (md2json.tokens(markdown)).map(token => ({[token.name]: token.text}));
```

**Comment by johnbeech** [comment]
Started a branch / PR:
- https://github.com/connected-web/md2json/pull/7

Not finished with support for all the cases.

@mmadson You're welcome to test from the branch to see if it handles your use case:

```sh
npm i -D github:connected-web/md2json#json2md-compatible-output
```

```js
const md2json = require('md2json')
const output = md2json({ markdown: '# Hello world', outputFormat: 'json2md' })
console.log('Output:', { output })
```

**Comment by mmadson** [comment]
WOW seriously impressed and appreciative. Will give your branch a try in a few days. Thanks again. 🙏

**Comment by johnbeech** [comment]
@mmadson I've added support for the majority of the md2json example cases now; e.g. images, code blocks, basic tables.

If you have chance to try, I'll probably end up merging and releasing 1.3.0 some time next week.

Updated README:
- https://github.com/connected-web/md2json/tree/json2md-compatible-output

**Comment by tomdyqin** [comment]
> @mmadson I've added support for the majority of the md2json example cases now; e.g. images, code blocks, basic tables.
> 
> If you have chance to try, I'll probably end up merging and releasing 1.3.0 some time next week.
> 
> Updated README:
> 
> * https://github.com/connected-web/md2json/tree/json2md-compatible-output

Wow, I'm looking forward to it. When can I experience 1.3.0？

**Comment by johnbeech** [comment]
@tomdyqin, @mmadson - Published 1.3.0 to npm just now:
- https://www.npmjs.com/package/@connected-web/md2json

Will close this issue; feel free to raise more!
